### ISSUE TITLE
measure db requets

### ISSUE THREAD

**Issue by iurimatias** [status-im/status-desktop#9507]
### Description

Conduct profiling of SQLite requests to determine if there are any bottlenecks in the processing queue for the database requests. The goal is to identify any areas of improvement for sequential processing by SQLite.

**Comment by John-44** [comment]
One question: we know that SQLite can only process queries sequentially.  **If** after profiling we see that queues of db queries are sometimes forming (as a result of SQLite only being able to process queries sequentially), then it might be worth investigating the feasibility of adding support to status-go to (optionally) use a db that is capable of processing queries in parallel like PostgreSQL.  However if this is a non-issue then we wouldn't want to waste time looking into using postgres with Status Go.

Another question: is the DB a contributor to any perf. issues in the app, and if so what are these issues and how could they be mitigated?  e.g. would there be a perf. gain to have seperate DBs per app (e.g one for wallet, one for chat, etc..) or even per community (thanks @iurimatias for mentioning this). Or are there any other DB perf optimisations like improving indexes, etc.. that we should look into.

Before we start doing any DB optimisation, it's important to first understand what aspects of DB performance today is contributing towards app perf issues (if any).  Once we understand this, we can then look into what optimisation could be solutions to these defined issues.

So tldr; of this issue is: understand all the db perf bottlenecks we have in the app when it's being used in the dogfooding context today.

**Comment by richard-ramos** [comment]
Any solution we produce should take in account that the mobile app also uses status-go, unless we want this performance measurement and possible solutions to improve performance of the DB to only focus on desktop. 

Perhaps these measurements have been done in mobile already? those could be used as a baseline to compare desktop performance against

**Comment by John-44** [comment]
> Any solution we produce should take in account that the mobile app also uses status-go, unless we want this performance measurement and possible solutions to improve performance of the DB to only focus on desktop.
> 
> Perhaps these measurements have been done in mobile already? those could be used as a baseline to compare desktop performance against

Ideally any db perf improvements would be beneficial for both mobile and desktop, however we shouldn't exclude the possibility that there may be some perf improvements that are only feasible or desirable on desktop, but not on mobile.  For example on Desktop we have the Discord import tool, and only Desktop clients can be owner nodes for communities (the Status Mobile client will never do these two things).  Therefor if we uncover db perf issues in any areas that are only relevant to Desktop and are not relevant for mobile, then we only need to look into ways of fixing these issues for Desktop.

**Comment by iurimatias** [comment]
possibly already done (Patryk)

**Comment by jrainville** [comment]
The DB calls have been improved a lot and the cause was mostly DB locking. Closing
